File: Daily/PY/Leetcode1646-获取生成数组中的最大值.py
class Solution:
    def getMaximumGenerated(self, n: int) -> int:
        if n == 0: return 0
        res = [0] * (n + 1)
        res[1] = 1
        for i in range(2,n + 1):
            res[i] = res[i // 2] + (i & 1) * res[i // 2 + 1]

        return max(res)

File: Daily/Rust/Leetcode1646-获取生成数组中的最大值.rs
impl Solution {
    pub fn get_maximum_generated(n: i32) -> i32 {
        if n == 0{
            return 0;
        }
        let mut res = vec![0;n as usize + 1];
        res[1] = 1;
        for i in 2..=n as usize{
            res[i] = res[i / 2] + (i & 1) * res[i / 2 + 1]; 
        }
        res[0..=n as usize].iter().max().unwrap().to_owned() as i32
    }
}